config/opal_load_platform.m4

```diff
@@ -97,6 +97,27 @@ AC_DEFUN([OPAL_LOAD_PLATFORM], [
             AC_SUBST(OPAL_PARAM_FROM_PLATFORM, "no")
         fi
 
+        patch_dir="${with_platform}.patches"
+        patch_done="${srcdir}/.platform_patches"
+        if [ test -d "${patch_dir}" ]; then
+            if [ test ! -f "${patch_done}" ]; then
+                OPAL_LOG_MSG([Applying patches from ${patch_dir}])
+                for one_patch in $patch_dir/*.patch ; do
+                    OPAL_LOG_MSG([Applying patch: $one_patch])
+                    patch -d ${srcdir} -p1 -t -s < ${one_patch}
+                    if [ test "$?" != "0" ]; then
+                        rm -f ${patch_done}
+                        AC_MSG_ERROR([Failed to apply patch ${one_patch}])
+                    fi
+                    touch ${patch_done}
```

Comment:
Mmm.  I see what you did there.

But it also gets touched if *one* patch is done, and others fail (even though you remove it if a failure occurs).

Perhaps a slightly cleaner way (that wouldn't be subject to Ctrl-C false-failures) could be:

```sh
patched=0
for one_patch ...
   OPAL_LOG_MSG(...)
   patch -d ...
   if test $? -ne 0; then
      AC_MSG_ERROR([...])
   fi
   patched=1
fi
if test patched -eq 1; then
    touch ${patch_done}
fi
```